src/app/container/container.component.html

```diff
@@ -0,0 +1,6 @@
+<div class="ekart--product--container">
+    <app-search (updatedSearchText)="textInput($event)"></app-search>
+    <app-product-list [searchText]="searchText" #productList></app-product-list>
+    <app-product-details *ngIf="productList.selectedProduct" [detailsProductListComponent]="productListComponent"></app-product-details>
```

Comment:
It is better to use <ng-content> to pass the component as a Child component and not to pass it as an Input